algorithms/unit_tests/TestSortByKey.hpp

```diff
@@ -0,0 +1,243 @@
+        sort_fails);
+
+    ASSERT_EQ(sort_fails, 0u);
+  }
+}
+
+TEST(TEST_CATEGORY, SortByKeyWithComparator) {
+  using ExecutionSpace = TEST_EXECSPACE;
+  using MemorySpace    = typename ExecutionSpace::memory_space;
+
+  ExecutionSpace space{};
+
+  SortImpl::Greater comparator;
+
+  for (auto keys_vector : {std::vector<int>{36, 19, 25, 17, 3, 7, 1, 2, 9},
+                           std::vector<int>{36, 19, 25, 17, 3, 9, 1, 2, 7},
+                           std::vector<int>{100, 19, 36, 17, 3, 25, 1, 2, 7},
+                           std::vector<int>{15, 5, 11, 3, 4, 8}}) {
+    auto const n = keys_vector.size();
+
+    auto keys = Kokkos::create_mirror_view_and_copy(
+        MemorySpace{},
+        Kokkos::View<int *, Kokkos::HostSpace, Kokkos::MemoryUnmanaged>(
+            keys_vector.data(), n));
+
+    auto keys_orig = Kokkos::create_mirror(space, keys);
+    Kokkos::deep_copy(space, keys_orig, keys);
+
+    Kokkos::View<int *, ExecutionSpace> permute("permute", n);
+    SortImpl::iota(space, permute);
+
+    Kokkos::Experimental::sort_by_key(space, keys, permute, comparator);
+
+    unsigned int sort_fails = 0;
+    Kokkos::parallel_reduce(
+        Kokkos::RangePolicy<ExecutionSpace>(space, 0, n),
+        SortImpl::is_sorted_by_key_struct<ExecutionSpace, decltype(keys),
+                                          decltype(permute), SortImpl::Greater>(
+            keys, keys_orig, permute, comparator),
+        sort_fails);
+
+    ASSERT_EQ(sort_fails, 0u);
+  }
+}
+
+TEST(TEST_CATEGORY, SortByKeyStaticExtents) {
+  using ExecutionSpace = TEST_EXECSPACE;
+
+  ExecutionSpace space{};
+
+  Kokkos::View<int[10], ExecutionSpace> keys("keys");
+
+  Kokkos::View<int[10], ExecutionSpace> values_static("values_static");
+  ASSERT_NO_THROW(
+      Kokkos::Experimental::sort_by_key(space, keys, values_static));
+
+  Kokkos::View<int *, ExecutionSpace> values_dynamic("values_dynamic", 10);
+  ASSERT_NO_THROW(
+      Kokkos::Experimental::sort_by_key(space, keys, values_dynamic));
+}
+
+template <typename ExecutionSpace, typename Keys, typename Values>
+void buildViewsForStrided(ExecutionSpace const &space, int n, Keys &keys,
+                          Values &values) {
+  Kokkos::parallel_for(
+      "create_data",
+      Kokkos::MDRangePolicy<Kokkos::Rank<3>, ExecutionSpace>(space, {0, 0, 0},
+                                                             {n, n, n}),
+      KOKKOS_LAMBDA(int i, int j, int k) {
+        keys(i, j, k)   = n - i;
+        values(i, j, k) = i;
+      });
+}
+
+TEST(TEST_CATEGORY, SortByKeyWithStrides) {
+  using ExecutionSpace = TEST_EXECSPACE;
+
+  ExecutionSpace space{};
+
+  auto const n = 10;
+
+  Kokkos::View<int ***, ExecutionSpace> keys("keys", n, n, n);
+  Kokkos::View<int ***, ExecutionSpace> values("values", n, n, n);
+  buildViewsForStrided(space, n, keys, values);
+
+  auto keys_sub   = Kokkos::subview(keys, Kokkos::ALL(), 0, 0);
+  auto values_sub = Kokkos::subview(values, Kokkos::ALL(), 0, 0);
```

Comment:
I would select extent `1` at least for one of the two views because layout left views would still be stride-1 with the current version
```C++
... = Kokkos::subview(values, Kokkos::ALL(), 1, 2);
... =Kokkos::subview(values, 4, Kokkos::ALL(), 6);